### PR TITLE
Add more to firmware signing key mix docs

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

File: lib/mix/tasks/nerves_hub.key.ex [hunk 1]
```diff
@@ -5,41 +5,68 @@ defmodule Mix.Tasks.NervesHub.Key do
...
+  development and production keys in use. For example, production devices
+  deployed with only the production public key will not accept firmware signed
+  by development keys.
+
+  To ensure that firmware includes keys registered with NervesHub, add the
+  following entries in your project's `config.exs`:
+
+    # Specify that NervesHub hub provides device provisioning metadata
+    config :nerves, :firmware,
+      provisioning: :nerves_hub
```
Comment (mobileoverlord) [inline lib/mix/tasks/nerves_hub.key.ex]: Adding provisioning is technically only required to get the overall connection certificates onto the device. I am unsure if it belongs in this part of the docs, but it should be stressed somewhere.

**Comment by mobileoverlord** [comment]
Other than that, it looks good

**Comment by fhunleth** [comment]
Updated

**Comment by fhunleth** [comment]
Hang on... I noticed a couple typos

**Comment by fhunleth** [comment]
Done

**Pull request merged**